src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -493,6 +501,79 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  /**
+   * @param snapshotSummary
+   * @param passports RAS passports as JWT tokens
+   * @return ValidatePassportResult indicating whether the snapshot's contents are accessible via
+   *     one of the supplied RAS passports
+   */
+  public ValidatePassportResult verifyPassportAuth(
+      SnapshotSummaryModel snapshotSummary, List<String> passports) {
+    if (passports.isEmpty()) {
+      throw new InvalidAuthorizationMethod("No RAS Passports supplied for accessing snapshot");
+    }
+    if (!SnapshotSummary.passportAuthorizationAvailable(snapshotSummary)) {
+      throw new InvalidAuthorizationMethod("Snapshot cannot be accessed by RAS Passports");
```

Comment:
Looks like this method gets called for all snapshots so it strikes me as odd to have this throw rather than just have this be a check in the calling method (e.g. don't even try to call `varifyPassportAuth` for non RAS snapshots)

Reply:
Presently, this method is called for snapshots only if SAM authentication fails first.

I found it tricky to strike the right balance in error messaging when integrating two auth approaches where we don't know for sure which one the user expects to work.  My instinct was that if we're falling back to this check, the user does want to know if the snapshot can't be accessed by RAS passports.  (But the empty passport list check is superfluous, I could get rid of that to streamline if you preferred.)

Reply:
I think it makes sense the way you have it.  It's a bummer...I'd love to test it out to get a sense of the UX...but, well, test accounts...